### DISCUSSION TITLE
formState.errors does not update on useForm.trigger

### DISCUSSION THREAD

**Discussion by JimmyRowland** [react-hook-form/react-hook-form#5843] [Q&A] [answered]
**Describe the bug**
formState.errors does not update on useForm.trigger.

**To Reproduce**

1. Go to [codesandbox](https://codesandbox.io/s/dry-pine-eyskx?file=/src/App.jsx)
![image](https://user-images.githubusercontent.com/16997894/124546911-2e0dc800-dde0-11eb-89fe-36f84f2f081f.png)

2. Click on YES radio button for the first radio group
![image](https://user-images.githubusercontent.com/16997894/124546932-3534d600-dde0-11eb-9125-66bf7e92e979.png)

3. Enter 222 in Any number input field and click out
![image](https://user-images.githubusercontent.com/16997894/124547001-51387780-dde0-11eb-8503-f7f0111f7a03.png)

Which will trigger setError on a hiddenNumberName input field

4. Enter 22 in Any number input field and click out
![image](https://user-images.githubusercontent.com/16997894/124547180-9b215d80-dde0-11eb-9522-9fcf60702f32.png)
Which will trigger setValue and trigger on  hiddenNumberName input field.

5. See bug. At this point, the error message should disappear, but error message only disappears on the next onBlur event.

**Codesandbox link (Required)**
https://codesandbox.io/s/dry-pine-eyskx?file=/src/App.jsx

**Expected behavior**
The error message should disappear when the user inputs 22 and click out.

**Desktop (please complete the following information):**

- OS: [Ubuntu 20.04]
- Browser [chrome]
- Version [90.0.4430.85]

**Additional context**
Calling trigger after the first validate event fixed the bug, but it would be nice to have a better solution.
![image](https://user-images.githubusercontent.com/16997894/124547777-81cce100-dde1-11eb-8f33-ec57a8447f06.png)

**Comment by bluebill1049** [comment, marked as answer]
thanks for the detailed issue report.

https://react-hook-form.com/api/useform/seterror

> https://react-hook-form.com/api/useform/seterror

<img width="961" alt="Screen Shot 2021-07-06 at 3 54 19 pm" src="https://user-images.githubusercontent.com/10513364/124549389-6f4eb800-de72-11eb-9af2-d157a9ee06f0.png">

**Reply by JimmyRowland** [reply]
Thanks for the quick answer, but the issue persists when the input field is not hidden
![image](https://user-images.githubusercontent.com/16997894/124551377-dd4d9d80-dde6-11eb-8a99-8314abff3a2a.png)

In this case, the error is associated with the input field "hiddenNumberName" right?

**Reply by JimmyRowland** [reply]
Also, the bug disappears when the second RadioGroup is removed
https://codesandbox.io/s/frosty-moon-53yio?file=/src/App.jsx

![image](https://user-images.githubusercontent.com/16997894/124552743-c90aa000-dde8-11eb-8314-8d2e10d575a3.png)

**Reply by bluebill1049** [reply]
`if (e.target.value >= 100) {` how would compare string with number?

**Reply by JimmyRowland** [reply]
`Number(e.target.value) >= 100`
After I fixed it, the validation still behaves the same.

**Reply by bluebill1049** [reply]
Can you simplify the example? it's hard to the real problem here... i suspect to do with async validation.

**Reply by JimmyRowland** [reply]
Does this work?
https://codesandbox.io/s/twilight-fog-d9b3p?file=/src/App.jsx

**Reply by bluebill1049** [reply]
This is much cleaner now, thanks will loo into this.

**Reply by JimmyRowland** [reply]
Thank you so much

**Reply by bluebill1049** [reply]
found the issue. issue created and will patch now.

**Reply by bluebill1049** [reply]
hold on: https://codesandbox.io/s/sleepy-tree-lxjvh?file=/src/App.jsx what's the error in here? 

it works for me.

**Reply by bluebill1049** [reply]
https://user-images.githubusercontent.com/10513364/124694921-89959e00-df25-11eb-9736-ef329f122dc1.mov

**Reply by JimmyRowland** [reply]
Hi Bill, the validation works without the required radio group.

But it takes 2 onBlur event to dismiss the error when there is a required radio group after the validated input field

https://codesandbox.io/s/twilight-fog-d9b3p?file=/src/App.jsx

https://user-images.githubusercontent.com/16997894/124794331-be90f900-df03-11eb-9b08-128630aa6c93.mp4

**Reply by bluebill1049** [reply]
I see, this has been resolved already then. will be in the next release

https://codesandbox.io/s/dreamy-oskar-6g9he?file=/src/App.jsx

**Reply by JimmyRowland** [reply]
Thanks
